### DISCUSSION TITLE
The question of the CLI

### DISCUSSION THREAD

**Discussion by nathanhazedawn** [karleeov/chatgpt-aoai#6] [Q&A] [answered]
How can you initiate the device code flow for authentication in the Azure CLI? 
What alternative authentication methods are recommended for production-level applications using the Azure CLI?
Please help QQQQ

**Comment by karleeov** [comment, marked as answer]
you can use the az login --use-device-code command. This will prompt you to enter the code displayed on the device and authenticate with your user credentials For production-level applications, it is recommended to use service principals for authentication. Service principals are a type of identity in Azure Active Directory (Azure AD) that can be used to authenticate applications and automate tasks in a secure manner. You can grant just the appropriate permissions needed to a service principal, keeping your automation secure
